From 7c1c8b1384e11166e1d3d1eb15750f3f8dd3241c Mon Sep 17 00:00:00 2001
From: Tom Stellard <thomas.stellard@amd.com>
Date: Wed, 06 Jun 2012 22:31:00 +0000
Subject: r600g: Fix reading vtx instruction offset from bytestream

---
diff --git a/src/gallium/drivers/r600/r600_shader.c b/src/gallium/drivers/r600/r600_shader.c
index 8a80dba..7ec1a1a 100644
--- a/src/gallium/drivers/r600/r600_shader.c
+++ b/src/gallium/drivers/r600/r600_shader.c
@@ -460,7 +460,7 @@ static int r600_vtx_from_byte_stream(struct r600_shader_ctx *ctx,
 	vtx.srf_mode_all = bytes[bytes_read++];
 	/* offset is 2 bytes wide */
 	vtx.offset = bytes[bytes_read++];
-	vtx.offset |= bytes[bytes_read++];
+	vtx.offset |= (bytes[bytes_read++] << 8);
 	vtx.endian = bytes[bytes_read++];
 
 	if (r600_bytecode_add_vtx(ctx->bc, &vtx)) {
--
cgit v0.9.0.2-2-gbebe
