Index: src/mesa/drivers/dri/r128/r128_state.c
===================================================================
RCS file: /cvs/mesa/Mesa/src/mesa/drivers/dri/r128/r128_state.c,v
retrieving revision 1.17
diff -p -u -r1.17 r128_state.c
--- src/mesa/drivers/dri/r128/r128_state.c	27 Oct 2005 21:21:05 -0000	1.17
+++ src/mesa/drivers/dri/r128/r128_state.c	20 Mar 2006 07:14:32 -0000
@@ -1248,6 +1248,7 @@ void r128DDInitState( r128ContextPtr rme
    rmesa->ClearColor = 0x00000000;
 
    switch ( rmesa->glCtx->Visual.depthBits ) {
+   case 0:
    case 16:
       rmesa->ClearDepth = 0x0000ffff;
       depth_bpp = R128_Z_PIX_WIDTH_16;
Index: src/mesa/drivers/dri/r200/r200_state_init.c
===================================================================
RCS file: /cvs/mesa/Mesa/src/mesa/drivers/dri/r200/r200_state_init.c,v
retrieving revision 1.27
diff -p -u -r1.27 r200_state_init.c
--- src/mesa/drivers/dri/r200/r200_state_init.c	7 Jan 2006 20:42:16 -0000	1.27
+++ src/mesa/drivers/dri/r200/r200_state_init.c	20 Mar 2006 07:14:33 -0000
@@ -176,6 +176,7 @@ void r200InitState( r200ContextPtr rmesa
    rmesa->state.color.clear = 0x00000000;
 
    switch ( ctx->Visual.depthBits ) {
+   case 0:
    case 16:
       rmesa->state.depth.clear = 0x0000ffff;
       rmesa->state.depth.scale = 1.0 / (GLfloat)0xffff;
Index: src/mesa/drivers/dri/r300/r300_state.c
===================================================================
RCS file: /cvs/mesa/Mesa/src/mesa/drivers/dri/r300/r300_state.c,v
retrieving revision 1.146
diff -p -u -r1.146 r300_state.c
--- src/mesa/drivers/dri/r300/r300_state.c	28 Feb 2006 23:08:28 -0000	1.146
+++ src/mesa/drivers/dri/r300/r300_state.c	20 Mar 2006 07:14:34 -0000
@@ -954,6 +954,7 @@ static void r300PolygonOffset(GLcontext 
 	GLfloat constant = units;
 	
 	switch (ctx->Visual.depthBits) {
+	case 0:
 	case 16:
 		constant *= 4.0;
 	break;
@@ -2036,6 +2037,7 @@ void r300ResetHwState(r300ContextPtr r30
 	r300->hw.unk4EA0.cmd[2] = 0xffffffff;
 
 	switch (ctx->Visual.depthBits) {
+	case 0:
 	case 16:
 		r300->hw.unk4F10.cmd[1] = R300_DEPTH_FORMAT_16BIT_INT_Z;
 	break;
@@ -2112,6 +2114,7 @@ void r300InitState(r300ContextPtr r300)
 	radeonInitState(&r300->radeon);
 
 	switch (ctx->Visual.depthBits) {
+	case 0:
 	case 16:
 		r300->state.depth.scale = 1.0 / (GLfloat) 0xffff;
 		depth_fmt = R200_DEPTH_FORMAT_16BIT_INT_Z;
Index: src/mesa/drivers/dri/radeon/radeon_state_init.c
===================================================================
RCS file: /cvs/mesa/Mesa/src/mesa/drivers/dri/radeon/radeon_state_init.c,v
retrieving revision 1.20
diff -p -u -r1.20 radeon_state_init.c
--- src/mesa/drivers/dri/radeon/radeon_state_init.c	7 Jan 2006 20:42:17 -0000	1.20
+++ src/mesa/drivers/dri/radeon/radeon_state_init.c	20 Mar 2006 07:14:34 -0000
@@ -173,6 +173,7 @@ void radeonInitState( radeonContextPtr r
    rmesa->state.color.clear = 0x00000000;
 
    switch ( ctx->Visual.depthBits ) {
+   case 0:
    case 16:
       rmesa->state.depth.clear = 0x0000ffff;
       rmesa->state.depth.scale = 1.0 / (GLfloat)0xffff;
