? r300-smooth-lines-1.diff
Index: r300_cmdbuf.c
===================================================================
RCS file: /cvs/mesa/Mesa/src/mesa/drivers/dri/r300/r300_cmdbuf.c,v
retrieving revision 1.53
diff -u -b -B -r1.53 r300_cmdbuf.c
--- src/mesa/drivers/dri/r300/r300_cmdbuf.c	20 Jan 2006 21:56:52 -0000	1.53
+++ src/mesa/drivers/dri/r300/r300_cmdbuf.c	7 Feb 2006 10:45:33 -0000
@@ -366,6 +366,8 @@
 		r300->hw.fpt.cmd[R300_FPT_CMD_0] = cmdpacket0(R300_PFS_TEXI_0, 0);
 	ALLOC_STATE( unk46A4, always, 6, "unk46A4", 0 );
 		r300->hw.unk46A4.cmd[0] = cmdpacket0(0x46A4, 5);
+	ALLOC_STATE( aa, always, R300_AA_CMDSIZE, "aa", 0 );
+		r300->hw.aa.cmd[R300_AA_CMD_0] = cmdpacket0(R300_ANTI_ALIAS, 0);
 	ALLOC_STATE( fpi[0], variable, R300_FPI_CMDSIZE, "fpi/0", 0 );
 		r300->hw.fpi[0].cmd[R300_FPI_CMD_0] = cmdpacket0(R300_PFS_INSTR0_0, 1);
 	ALLOC_STATE( fpi[1], variable, R300_FPI_CMDSIZE, "fpi/1", 1 );
Index: r300_context.h
===================================================================
RCS file: /cvs/mesa/Mesa/src/mesa/drivers/dri/r300/r300_context.h,v
retrieving revision 1.84
diff -u -b -B -r1.84 r300_context.h
--- src/mesa/drivers/dri/r300/r300_context.h	27 Jan 2006 13:21:47 -0000	1.84
+++ src/mesa/drivers/dri/r300/r300_context.h	7 Feb 2006 10:45:33 -0000
@@ -339,6 +339,9 @@
 #define R300_FPT_INSTR_0	1
 #define R300_FPT_CMDSIZE	65
 
+#define R300_AA_CMD_0		0
+#define R300_AA_CMDSIZE	1
+
 #define R300_FPI_CMD_0		0
 #define R300_FPI_INSTR_0	1
 #define R300_FPI_CMDSIZE	65
@@ -446,6 +449,7 @@
 	struct r300_state_atom fp;	/* fragment program cntl + nodes (4600) */
 	struct r300_state_atom fpt;     /* texi - (4620) */
 	struct r300_state_atom unk46A4;	/* (46A4) */
+	struct r300_state_atom aa;	/* anti-aliasing - (46B4) */
 	struct r300_state_atom fpi[4];	/* fp instructions (46C0/47C0/48C0/49C0) */
 	struct r300_state_atom unk4BC0;	/* (4BC0) */
 	struct r300_state_atom unk4BC8;	/* (4BC8) */
Index: r300_reg.h
===================================================================
RCS file: /cvs/mesa/Mesa/src/mesa/drivers/dri/r300/r300_reg.h,v
retrieving revision 1.54
diff -u -b -B -r1.54 r300_reg.h
--- src/mesa/drivers/dri/r300/r300_reg.h	27 Jan 2006 13:21:47 -0000	1.54
+++ src/mesa/drivers/dri/r300/r300_reg.h	7 Feb 2006 10:45:35 -0000
@@ -916,6 +916,12 @@
 #			define R300_FPITX_OP_TXP			3
 #			define R300_FPITX_OP_TXB			4
 
+/* Probably some other stuff in this register too, involved with tex/blend
+ * enabling, specular enable, fog enable, alpha test enable on r200 */
+/* Other possibility: 0x453c? */
+#define R300_ANTI_ALIAS						0x46B4
+#		define R300_ANTI_ALIAS_LINE				(1 << 0)
+
 /* ALU
 // The ALU instructions register blocks are enumerated according to the order
 // in which fglrx. I assume there is space for 64 instructions, since
Index: r300_state.c
===================================================================
RCS file: /cvs/mesa/Mesa/src/mesa/drivers/dri/r300/r300_state.c,v
retrieving revision 1.144
diff -u -b -B -r1.144 r300_state.c
--- src/mesa/drivers/dri/r300/r300_state.c	27 Jan 2006 14:35:24 -0000	1.144
+++ src/mesa/drivers/dri/r300/r300_state.c	7 Feb 2006 10:45:37 -0000
@@ -450,6 +450,15 @@
 		update_early_z(ctx);
 		break;
 
+	case GL_LINE_SMOOTH:
+		R300_STATECHANGE( r300, aa );
+		if (state) {
+			r300->hw.aa.cmd[R300_AA_CMD_0] |=  R300_ANTI_ALIAS_LINE;
+		} else {
+			r300->hw.aa.cmd[R300_AA_CMD_0] &= ~R300_ANTI_ALIAS_LINE;
+		}
+		break;
+
 	case GL_STENCIL_TEST:
 		if (r300->state.stencil.hw_stencil) {
 			R300_STATECHANGE(r300, zs);
