Index: r300_reg.h
===================================================================
RCS file: /cvs/mesa/Mesa/src/mesa/drivers/dri/r300/r300_reg.h,v
retrieving revision 1.54
diff -u -b -B -u -r1.54 r300_reg.h
--- src/mesa/drivers/dri/r300/r300_reg.h	27 Jan 2006 13:21:47 -0000	1.54
+++ src/mesa/drivers/dri/r300/r300_reg.h	7 Feb 2006 12:49:26 -0000
@@ -350,7 +350,7 @@
 #	define R300_GB_LINE_STUFF_ENABLE	(1<<1)
 #	define R300_GB_TRIANGLE_STUFF_ENABLE	(1<<2)
 #	define R300_GB_STENCIL_AUTO_ENABLE	(1<<4)
-#	define R300_GB_UNK30			(1<<30)
+#	define R300_GB_UNK30			(1<<31)
 	/* each of the following is 2 bits wide */
 #define R300_GB_TEX_REPLICATE	0
 #define R300_GB_TEX_ST		1
Index: r300_render.c
===================================================================
RCS file: /cvs/mesa/Mesa/src/mesa/drivers/dri/r300/r300_render.c,v
retrieving revision 1.100
diff -u -b -B -u -r1.100 r300_render.c
--- src/mesa/drivers/dri/r300/r300_render.c	27 Jan 2006 14:09:05 -0000	1.100
+++ src/mesa/drivers/dri/r300/r300_render.c	7 Feb 2006 12:49:26 -0000
@@ -730,6 +730,9 @@
 
 //	LOCK_HARDWARE(&(rmesa->radeon));
 
+	reg_start(R300_GB_SELECT,0);
+	e32(0x00000010);
+	
 	reg_start(R300_RB3D_DSTCACHE_CTLSTAT,0);
 	e32(0x0000000a);
 
@@ -802,7 +805,7 @@
 	FALLBACK_IF(ctx->Line.StippleFlag);
 
 	/* HW doesnt appear to directly support these */
-	FALLBACK_IF(ctx->Line.SmoothFlag); // GL_LINE_SMOOTH
+	//FALLBACK_IF(ctx->Line.SmoothFlag); // GL_LINE_SMOOTH
 	FALLBACK_IF(ctx->Point.SmoothFlag); // GL_POINT_SMOOTH
 	/* Rest could be done with vertex fragments */
 	if (ctx->Extensions.NV_point_sprite || ctx->Extensions.ARB_point_sprite)
Index: r300_state.c
===================================================================
RCS file: /cvs/mesa/Mesa/src/mesa/drivers/dri/r300/r300_state.c,v
retrieving revision 1.144
diff -u -b -B -u -r1.144 r300_state.c
--- src/mesa/drivers/dri/r300/r300_state.c	27 Jan 2006 14:35:24 -0000	1.144
+++ src/mesa/drivers/dri/r300/r300_state.c	7 Feb 2006 12:49:26 -0000
@@ -1147,7 +1147,7 @@
 			/* Currently disabled! */
 			r300->hw.tex.unknown1.cmd[R300_TEX_VALUE_0 + hw_tmu] = 0x0;
 			r300->hw.tex.size.cmd[R300_TEX_VALUE_0 + hw_tmu] = t->size;
-			r300->hw.tex.format.cmd[R300_TEX_VALUE_0 + hw_tmu] = t->format;
+			r300->hw.tex.format.cmd[R300_TEX_VALUE_0 + hw_tmu] = 0x02000a00; // t->format | 0x2000000;
 			r300->hw.tex.pitch.cmd[R300_TEX_VALUE_0 + hw_tmu] = t->pitch_reg;
 			r300->hw.tex.offset.cmd[R300_TEX_VALUE_0 + hw_tmu] = t->offset;
 			r300->hw.tex.unknown4.cmd[R300_TEX_VALUE_0 + hw_tmu] = 0x0;
@@ -1882,7 +1882,7 @@
 
 	r300->hw.gb_enable.cmd[1] = R300_GB_POINT_STUFF_ENABLE
 		| R300_GB_LINE_STUFF_ENABLE
-		| R300_GB_TRIANGLE_STUFF_ENABLE /*| R300_GB_UNK30*/;
+		| R300_GB_TRIANGLE_STUFF_ENABLE | R300_GB_UNK30;
 
 	r300->hw.gb_misc.cmd[R300_GB_MISC_MSPOS_0] = 0x66666666;
 	r300->hw.gb_misc.cmd[R300_GB_MISC_MSPOS_1] = 0x06666666;
