From cbf4fec52960ffd74b78ea45d0b917c76f553ebf Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Kristian=20H=C3=B8gsberg?= <krh@redhat.com>
Date: Mon, 1 Dec 2008 20:53:17 -0500
Subject: [PATCH] Bump dri2proto requirement to 1.99.3.

---
 configure.ac |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/configure.ac b/configure.ac
index 8a02f27..3c3fd36 100644
--- a/configure.ac
+++ b/configure.ac
@@ -855,7 +855,7 @@ if test "x$DRI" = xyes; then
 	AC_SUBST(GL_CFLAGS)
 fi
 
-PKG_CHECK_MODULES([DRI2PROTO], [dri2proto >= 1.99.1],
+PKG_CHECK_MODULES([DRI2PROTO], [dri2proto >= 1.99.3],
                   [HAVE_DRI2PROTO=yes], [HAVE_DRI2PROTO=no])
 case "$DRI2,$HAVE_DRI2PROTO" in
 	yes,no)
-- 
1.6.0.4

