From b2a3d165b09b0cf89417ef7d824a529dd4572399 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Cardona?= <remi@gentoo.org>
Date: Sat, 3 Oct 2009 12:43:48 +0200
Subject: [PATCH 2/2] Gentoo-specific customizations

---
 xinitrc.cpp |   11 ++++++-----
 1 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/xinitrc.cpp b/xinitrc.cpp
index 5643340..6687441 100644
--- a/xinitrc.cpp
+++ b/xinitrc.cpp
@@ -3,8 +3,8 @@ XCOMM $Xorg: xinitrc.cpp,v 1.3 2000/08/17 19:54:30 cpqbld Exp $
 
 userresources=$HOME/.Xresources
 usermodmap=$HOME/.Xmodmap
-sysresources=XINITDIR/.Xresources
-sysmodmap=XINITDIR/.Xmodmap
+sysresources=XINITDIR/Xresources
+sysmodmap=XINITDIR/Xmodmap
 
 XCOMM merge in defaults and keymaps
 
@@ -85,8 +85,6 @@ fi
 XCOMM This is the fallback case if nothing else is executed above
 #endif /* !defined(__SCO__)  && !defined(__UNIXWARE__) */
 
-#ifdef __APPLE__
-
 if [ -d XINITDIR/xinitrc.d ] ; then
 	for f in XINITDIR/xinitrc.dXSLASHGLOB.sh ; do
 		[ -x "$f" ] && . "$f"
@@ -94,7 +92,10 @@ if [ -d XINITDIR/xinitrc.d ] ; then
 	unset f
 fi
 
-#endif
+if [ -n "`/etc/X11/chooser.sh`" ]; then
+	exec "`/etc/X11/chooser.sh`"
+	return
+fi
 
 TWM &
 XCLOCK -geometry 50x50-1+1 &
-- 
1.6.5.rc2

