From 8c3e2166379f434ac53aaefae7f8d9d102c1226b Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Fri, 24 Oct 2008 14:47:27 -0400
Subject: [PATCH] Quirk: No LVDS on Dell Studio Hybrid

---
 src/i830_quirks.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/src/i830_quirks.c b/src/i830_quirks.c
index 12cab86..8dae235 100644
--- a/src/i830_quirks.c
+++ b/src/i830_quirks.c
@@ -255,6 +255,8 @@ static i830_quirk i830_quirk_list[] = {
     { PCI_CHIP_I965_GM, 0x1028, 0x0286, quirk_ignore_tv },
     /* Dell Vostro A840 (LP: #235155) */
     { PCI_CHIP_I965_GM, 0x1028, 0x0298, quirk_ignore_tv },
+    /* Dell Studio Hybrid */
+    { PCI_CHIP_I965_GM, 0x1028, 0x0279, quirk_ignore_lvds },
 
     /* Lenovo Napa TV (use dmi)*/
     { PCI_CHIP_I945_GM, 0x17aa, SUBSYS_ANY, quirk_lenovo_tv_dmi },
-- 
1.6.0.3

