From 7f0768ff9831e16644dacf98dc74703eba050553 Mon Sep 17 00:00:00 2001
From: Zhenyu Wang <zhenyu.z.wang@intel.com>
Date: Mon, 3 Nov 2008 14:51:14 +0800
Subject: [PATCH] SDVO: fix wrong order of sdvo version's major/minor

---
 src/i830_sdvo_regs.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/i830_sdvo_regs.h b/src/i830_sdvo_regs.h
index 4d55555..be35b8d 100644
--- a/src/i830_sdvo_regs.h
+++ b/src/i830_sdvo_regs.h
@@ -50,8 +50,8 @@ struct i830_sdvo_caps {
     uint8_t vendor_id;
     uint8_t device_id;
     uint8_t device_rev_id;
-    uint8_t sdvo_version_major;
     uint8_t sdvo_version_minor;
+    uint8_t sdvo_version_major;
     unsigned int sdvo_input_count:2;
     unsigned int smooth_scaling:1;
     unsigned int sharp_scaling:1;
-- 
1.6.0.3

