--- a/libxcb/src/xcbint.h	2008-11-01 12:18:48.000000000 -0400
+++ b/libxcb/src/xcbint.h	2008-11-01 12:02:02.241276875 -0400
@@ -80,6 +80,11 @@
     pthread_cond_t cond;
     int writing;
 
+    pthread_cond_t socket_cond;
+    void (*return_socket)(void *closure);
+    void *socket_closure;
+    int socket_moving;
+
     char queue[XCB_QUEUE_BUFFER_SIZE];
     int queue_len;
 
@@ -107,11 +112,6 @@
     pthread_cond_t event_cond;
     int reading;
 
-    pthread_cond_t socket_cond;
-    void (*return_socket)(void *closure);
-    void *socket_closure;
-    int socket_moving;
-
     char queue[XCB_QUEUE_BUFFER_SIZE];
     int queue_len;
 
